platoon/channel/worker.py

```diff
@@ -0,0 +1,569 @@
+        else:
+            raise AttributeError("pygpu or theano is not imported")
+
+    def init_mb_sock(self, port, hwm=10):
+        """
+        Initialize the mini-batch socket.
+
+        This must be called before using :meth:`recv_mb`.
+
+        Parameters
+        ----------
+        port : int
+            The port to reach the mini-batch server on.
+        hwm : int
+            High water mark, see pyzmq docs.
+
+        """
+        self.asocket = self.context.socket(zmq.PULL)
+        self.asocket.setsockopt(zmq.LINGER, 0)
+        self.asocket.set_hwm(hwm)
+        self.asocket.connect("tcp://localhost:{}".format(port))
+
+        self.apoller = zmq.Poller()
+        self.apoller.register(self.asocket, zmq.POLLIN)
+
+    def _init_control_socket(self, port):
+        """
+        Initialize control socket.
+
+        This must be called before using :meth:`send_req`.
+
+        Parameters
+        ---------
+        port : int
+            Port where the control master is listening on.
+
+        """
+        self.csocket = self.context.socket(zmq.REQ)
+        self.csocket.setsockopt(zmq.LINGER, 0)
+        self.csocket.connect('tcp://localhost:{}'.format(port))
+
+        self.cpoller = zmq.Poller()
+        self.cpoller.register(self.csocket, zmq.POLLIN)
+
+################################################################################
+#                            Collectives Interface                             #
+################################################################################
+
+    def shared(self, array):
+        """Creates a new posix shared memory buffer to be shared among Workers
+        and their Controller and maps the size of `array` to that buffer.
+
+        Controller is requested to create a new shared memory buffer with the
+        same size as `array` in order to be used in multi-node/gpu platoon
+        collective operations through the new worker interface. All participants
+        in a host have access to that memory.
+
+        :param array: will correspond to shared buffer in host with the same size
+        :type array: :ref:`pygpu.gpuarray.GpuArray`
+
+        Notes
+        -----
+        For internals: There should probably be a barrier across hosts' worker
+        here to ensure that so far, controller has serviced all workers (if
+        needed) a new shared memory's name. This is due to the fact that
+        Controller can service one Worker at a time and a platoon collective
+        service is a blocking one across Controllers. Current implementation
+        is valid because calls to pygpu.collectives interface are synchronous
+        across workers.
+
+        """
+        if not isinstance(array, pygpu.gpuarray.GpuArray):
+            raise TypeError("`array` input is not pygpu.gpuarray.GpuArray.")
+
+        bytesize = array.size * array.itemsize
+
+        if bytesize in self.shared_arrays:
+            return self.shared_arrays[bytesize]
```

Comment:
Your explanation makes sense, but I the code isn't clear at all about this.  Maybe add a small comment saying this above the line.